servers/src/main/java/tachyon/worker/block/BlockServiceHandler.java

```diff
@@ -0,0 +1,204 @@
+   * folder will be returned.
+   *
+   * @param userId
+   */
+  public String getUserUfsTempFolder(long userId) throws TException {
+    return mWorker.getUserUfsTmpFolder(userId);
+  }
+
+  /**
+   * Lock the file in Tachyon's space while the user is reading it, and the path of the block file
+   * locked will be returned, if the block file is not found, FileDoesNotExistException will be
+   * thrown.
+   *
+   * @param blockId
+   * @param userId
+   */
+  public String lockBlock(long blockId, long userId) throws TException {
+    try {
+      long lockId = mWorker.lockBlock(userId, blockId);
+      return mWorker.readBlock(userId, blockId, lockId);
+    } catch (IOException ioe) {
+      throw new FileDoesNotExistException("Block " + blockId + " does not exist on this worker.");
+    }
+  }
+
+  /**
+   * Used to promote block on under storage layer to top storage layer when there are more than one
+   * storage layers in Tachyon's space. return true if the block is successfully promoted, false
+   * otherwise.
+   *
+   * @param blockId
+   */
+  public boolean promoteBlock(long blockId) throws TException {
+    try {
+      // TODO: Maybe add constant location for First Tier?
+      mWorker.moveBlock(Users.MIGRATE_DATA_USER_ID, blockId, 1);
+      return true;
+    } catch (IOException ioe) {
+      throw new TException(ioe);
+    }
+  }
+
+  /**
+   * Used to allocate location and space for a new coming block, worker will choose the appropriate
+   * storage directory which fits the initial block size by some allocation strategy, and the
+   * temporary file path of the block file will be returned. if there is no enough space on Tachyon
+   * storage OutOfSpaceException will be thrown, if the file is already being written by the user,
+   * FileAlreadyExistException will be thrown.
+   *
+   * @param userId
+   * @param blockId
+   * @param initialBytes
+   */
+  public String requestBlockLocation(long userId, long blockId, long initialBytes)
+      throws TException {
+    try {
+      // NOTE: right now, we ask allocator to allocate new blocks in MEM tier by setting location
+      // to be 1.
+      return mWorker.createBlock(userId, blockId, 1, initialBytes);
+    } catch (IOException ioe) {
+      throw new OutOfSpaceException("Failed to allocate " + initialBytes + " for user " + userId);
+    }
+  }
+
+  /**
+   * Used to request space for some block file. return true if the worker successfully allocates
+   * space for the block on block’s location, false if there is no enough space, if there is no
+   * information of the block on worker, FileDoesNotExistException will be thrown.
+   *
+   * @param userId
+   * @param blockId
+   * @param requestBytes
+   */
+  public boolean requestSpace(long userId, long blockId, long requestBytes) throws TException {
+    try {
+      mWorker.requestSpace(userId, blockId, requestBytes);
+      return true;
+    } catch (IOException ioe) {
+      return false;
```

Comment:
Since we hide exception in here, better to at least `LOG.error` the exception, rather than letting it proceed silently.